bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -410,15 +418,19 @@ protected CallOptions getOperationCallOptions() {
    *     reads.
    * @return a {@link CallOptions}
    */
-  protected CallOptions getRpcCallOptions() {
-    if (callOptions.getDeadline() != null || isStreamingRead()) {
+  protected CallOptions getRpcAttemptCallOptions() {
+    if (callOptionsFactory.isRpcAttemptTimeoutEnabled(request)) {
+      return callOptionsFactory.createForRpcAttempt(request, operationCallOptions.getDeadline());
+    }
+
+    if (operationCallOptions.getDeadline() != null || isStreamingRead()) {
       // If the user set a deadline, honor it.
       // If this is a streaming read, then the Watchdog will take affect and ensure that hanging
       // does not occur.
       return getOperationCallOptions();
     } else {
       // Unary calls should fail after 6 minutes, if there isn't any response from the server.
-      return callOptions.withDeadlineAfter(UNARY_DEADLINE_MINUTES, TimeUnit.MINUTES);
+      return operationCallOptions.withDeadlineAfter(UNARY_DEADLINE_MINUTES, TimeUnit.MINUTES);
```

Comment:
Lets move this into CallOptionsFactory to centralize all of the deadline computations

Reply:
Refactor is done